devito/visitors.py

```diff
@@ -179,17 +185,34 @@ def visit_Expression(self, o, ret=None, queue=None):
         ret.setdefault(key, []).append(o)
         return ret
 
+    visit_Element = visit_Expression
+
 
 class FindSymbols(Visitor):
 
     @classmethod
     def default_retval(cls):
         return []
 
-    """Find all :class:`SymbolicData` and :class:`IndexedData`
-    instances in an Iteration/Expression tree.
+    """Find symbols in an Iteration/Expression tree.
+
+    :param mode: Drive the search for symbols. Accepted values are: ::
+
+        * 'with-data' (default): all :class:`SymbolicData` and :class:`IndexedData`
+          instances are collected.
+        * 'free-symbols': all free symbols appearing in :class:`Expression` are
+          collected.
```

Comment:
Ok, so we technically also return `Dimension` objects to denote open-loop sizes. Admittedly this is a bit hacky, so we should either mention it in the docstring, or maybe even have a special mode for this? I suspect this will get touched up further once we do more blocking/tiling.